Complete test cases for the scenario in which there is only 1 and 2 page of result, validate the result and pagination

nthikimngoc opened the pull request.

nthikimngoc committed:
.

liamnguyeniaa commented on src/homework-NN.js:
review the comments left on Nhi's PR and add something different than Nhi's approach.

liamnguyeniaa reviewed (changes requested):
comments

nthikimngoc committed:
edit homework on Wednesday, "search for "tuyển sinh" and validate some results appear"

nthikimngoc committed:
validate Click Next Page multiple times

nthikimngoc committed:
Homework 02 - Due on next Monday, 01/15/2024
Write test cases for the scenario in which there is only 1 and 2 page of result

liamnguyeniaa commented on src/02-so2-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,18 @@
+  
+    await t
+      .expect(Selector(".main .thong-tin .blog-details .btn").exists)
+      .ok("At more six search result should exist");
```
incorrect message. 

need to validate that the pagination doesn't appear.

liamnguyeniaa commented on src/02-so3-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,31 @@
+
+test("search for 'Công khai' and there are 2 pages of result, validate the result and pagination", async (t) => {
+  const searchInputSelector = Selector(".search .input-group .search");
+  const ClickNextPage = Selector(".main .page-item:last-child .page-link");
```
incorrect variable name. use camelCase.

liamnguyeniaa commented on src/02-so3-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,31 @@
+
+  await t
+    .expect(Selector(".main .thong-tin .blog-details .btn").exists)
+    .ok("At more six search result should exist");
```
incorrect message

liamnguyeniaa commented on src/02-so3-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,31 @@
+    .expect(Selector(".main .thong-tin .blog-details .btn").exists)
+    .ok("At more six search result should exist");
+
+  for (let i = 0; i < 5; i++) {
```
there are only 2 pages. Why do we need to click 5 times?

liamnguyeniaa commented on src/02-so3-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,31 @@
+
+  await t.click(ClickNextPage);
+
+  await t.click(ClickPreviousPage);
```
the test case should never end with action.

liamnguyeniaa commented on src/02-ttts2023-agu-pagination-ngocnguyen.test.js:
```diff
@@ -0,0 +1,51 @@
+
+test("search for 'tuyển sinh' and validate some results appear", async (t) => {
+  const searchInputSelector = Selector(".search .input-group .search");
+  const ClickNextPage = Selector(".main .page-item:last-child .page-link");
```
incorrect variable name.

nthikimngoc replied:
Done

nthikimngoc replied:
Done

liamnguyeniaa reviewed (commented):
1. Review variable naming
1. make sure the test case never end with action
2. fix all the comments.

aduyng committed:
Merge branch 'develop' into NgocNguyen-homework.js

nthikimngoc committed:
edited variable name, Based on Kieu Hau's code to compare between 2 pages

nthikimngoc committed:
edited message and validate that the pagination doesn't appear

nthikimngoc committed:
edited message, edited variable name, validate result and pagination with there are 2 pages of result

nthikimngoc committed:
edited (delete messege is wrong)

nthikimngoc committed:
edited, message name
